### PR TITLE
=str Make use of OptionVal in Recover operator.

### PR DESCRIPTION
**Pull request by He-Pin**
Change to use OptionVal instead.

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala [hunk 1]
```diff
@@ -280,33 +280,28 @@ private[stream] object Collect {
 
       import Collect.NotApplied
 
-      var recovered: Option[T] = None
+      var recovered: OptionVal[T] = OptionVal.none
 
-      override def onPush(): Unit = {
-        push(out, grab(in))
-      }
+      override def onPush(): Unit = push(out, grab(in))
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala:285]: we typically prefer a newline here, but not important
(the `{` is indeed superfluous)

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
@He-Pin can you rebase this, and then we can merge

**Pull request merged**